### PR TITLE
Fix Sharing of long goo.gl links

### PR DESCRIPTION
**Pull request by nishadg246**
Since goo.gl doesn't reshorten existing goo.gl links, we have to find
the final url and reshorten it.

### CODE REVIEW

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java [hunk 1]
```diff
@@ -102,42 +102,74 @@ public void onCreate() {
...
+          }
+          @Override
+          public void onError(String oldUrl) {
+            Toast.makeText(getApplicationContext(), getString(R.string.shorten_error),
+                Toast.LENGTH_LONG).show();
+          }
+        };
+        UrlShortenerClient shortenerClient = UrlShortenerClient.getInstance(this);
+
+        if (mDisplayUrl.contains("goo.gl")) {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java]: Could you split this out into its own method?  This is getting pretty long.

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java [hunk 2]
```diff
@@ -101,41 +106,79 @@ public void onCreate() {
...
+          shortenerClient.shortenUrl(mDisplayUrl, urlSetter, TAG);
+        }            
+      }
+      return START_STICKY;
+    }
+
+    private void fetchAndShorten(UrlShortenerClient shortenerClient,
+        UrlShortenerClient.ShortenUrlCallback urlSetter) {
+      PwsClient pwsClient = new PwsClient();
+      pwsClient.resolve(Arrays.asList(mDisplayUrl),new PwsResultCallback() {
```
Comment (nondebug) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java]: space after comma

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java [hunk 3]
```diff
@@ -101,41 +106,79 @@ public void onCreate() {
...
+        @Override
+        public void onPwsResult(PwsResult pwsResult) {
+          Log.d(TAG,"success");
```
Comment (nondebug) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java]: space after comma (also two more for the other Log.d calls)

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java [hunk 4]
```diff
@@ -17,6 +17,9 @@
 package org.physical_web.physicalweb;
 
 import org.physical_web.physicalweb.ble.AdvertiseDataUtils;
+import org.physical_web.collection.PwsClient;
```
Comment (nondebug) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java]: imports should be in ASCII order, so move these org.physical_web.collection.* imports above org.physical_web.physicalweb.*

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java [hunk 5]
```diff
@@ -101,41 +106,79 @@ public void onCreate() {
...
+          Log.d(TAG,"success");
+          shortenerClient.shortenUrl(pwsResult.getSiteUrl(), urlSetter, TAG);
         }
-        return START_STICKY;
+
+        @Override
+        public void onPwsResultAbsent(String url) {
+          Log.d(TAG,"fail absent");
+          Toast.makeText(getApplicationContext(), getString(R.string.shorten_error),
+            Toast.LENGTH_LONG).show();
```
Comment (nondebug) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java]: indent two levels for a line continuation (here and in onPwsResultError below)

**Comment by nondebug** [comment]
lgtm after addressing comments

**Comment by nondebug** [comment]
lgtm

**Pull request merged**